packages/worker/src/runtime.ts

```diff
@@ -284,13 +286,36 @@ export class Runtime {
   }
 
   /** @hidden */
-  public async createReplayWorker(options: native.WorkerOptions, history: History): Promise<native.Worker> {
-    return await this.createNative(
-      promisify(native.newReplayWorker),
-      this.native,
-      options,
-      byteArrayToBuffer(temporal.api.history.v1.History.encodeDelimited(history).finish())
-    );
+  public async createReplayWorker(options: native.WorkerOptions): Promise<native.ReplayWorker> {
+    return await this.createNativeNoBackRef(async () => {
+      const fn = promisify(native.newReplayWorker);
+      const replayWorker = await fn(this.native, options);
+      this.backRefs.add(replayWorker.worker);
+      return replayWorker;
+    });
+  }
+
+  /**
+   * Push history to a replay worker's history pusher stream.
+   *
+   * Hidden in the docs because it is only meant to be used internally by the Worker.
+   *
+   * @hidden
+   */
+  public async pushHistory(pusher: native.HistoryPusher, workflowId: string, history: History): Promise<void> {
```

Comment:
This seems a little bit out of place... Why bring this here rather than doing it directly in Worker?